src/mlpack/core/optimizers/adam/adam.hpp

```diff
@@ -107,7 +110,7 @@ class AdamType
    * @return Objective value of the final point.
    */
   template<typename DecomposableFunctionType>
-  double Optimize(DecomposableFunctionType& function, arma::mat& iterate)
+  double Optimize(DecomposableFunctionType& function, MatrixType& iterate)
```

Comment:
I think we could write:

```c++
template<typename DecomposableFunctionType, typename MatType>
double Optimize(DecomposableFunctionType& function, MatType& iterate)
{
...
}
```

That way, we can avoid the extra class template parameter. What do you think?

Note, I just used ``MatType`` instead of ``MatrixType`` for consistency reasons.